src/controller.ts

```diff
@@ -1,23 +1,73 @@
 import { Application } from "pixi.js";
 import { Minigame, MinigameDelegate } from "./minigames/minigame";
 import { KeyboardMinigame } from "./minigames/keyboardMinigame";
+import { Interlude, InterludeDelegate } from "./interlude/interlude";
 
+const ALL_MINIGAMES = [Minigame, KeyboardMinigame];
 /** Controls the flow of the game. */
-export class Controller implements MinigameDelegate {
-  private currentMinigame?: Minigame;
+export class Controller implements MinigameDelegate, InterludeDelegate {
+  private completedMinigamePhases = 0;
+
+  private minigameQueue: Minigame[] = [];
+  private minigameWinCount = 0;
+
+  private currentScreen?: Minigame | Interlude;
 
   constructor(private readonly app: Application) { }
 
   start() {
-    this.currentMinigame = new KeyboardMinigame(this.app, this);
-    this.currentMinigame.attach();
+    this.populateMinigameQueue(10);
+    this.startNextMinigame();
   }
 
   onMinigameEnd(passed: boolean) {
-    this.currentMinigame!.detach();
-    this.currentMinigame = undefined;
-    console.log('Player', passed ? 'won' : 'lost');
-    // For now just restart the same minigame again.
-    this.start();
+    this.currentScreen!.detach();
+    if (passed) {
+      this.minigameWinCount++;
+    }
+    console.log(passed ? 'Nice!' : 'Too bad!', 'Player has won', this.minigameWinCount);
+
+    if (this.minigameWinCount >= 5) {
+      // Player has met the requirements to proceed to the next phase.
+      this.completedMinigamePhases++;
+      if (this.completedMinigamePhases >= 3) {
+        console.log('You are the power lifting champion!');
+        return;
+      }
+
+      this.currentScreen = new Interlude(this.app, this);
+      this.currentScreen.attach();
+      this.minigameWinCount = 0;
+      return;
+    }
+
+    if (this.minigameQueue.length === 0) {
+      // Player has run out of chances to proceed to the next phase.
+      console.log('You lose!');
+      return;
+    }
+
+    this.startNextMinigame();
+  }
+
+  onInterludeEnd() {
+    this.currentScreen!.detach();
+
+    this.populateMinigameQueue(10);
+    this.startNextMinigame();
+  }
+
+  private populateMinigameQueue(count: number, pool = ALL_MINIGAMES) {
+    // Reset the current queue.
+    this.minigameQueue.length = 0;
+    for (let i = 0; i < count; i++) {
+      const minigame = new pool[Math.floor(Math.random() * pool.length)](this.app, this);
+      this.minigameQueue.push(minigame);
+    }
+  }
+
+  private startNextMinigame() {
```

Comment:
Maybe just put currentScreen.detach here?

Reply:
We might want to detach without starting a new minigame, e.g. when going to an interlude. Added a safety detach here anyway